packages/core/strapi/lib/services/entity-service/types/params/populate.d.ts

```diff
@@ -39,7 +40,37 @@ type StringNotation<TSchemaUID extends Common.UID.Schema> =
  */
 type ArrayNotation<TSchemaUID extends Common.UID.Schema> = StringNotation<TSchemaUID>[];
 
+type PopulateDynamicZoneFragments<TSchemaUID extends Common.UID.Schema> = {
+  [TKey in Attribute.GetKeysByType<TSchemaUID, 'dynamiczone'>]: {
+    on: {
+      [TComponentUID in GetComponentsFromDynamicZone<TSchemaUID, TKey>[number]]?:
+        | Boolean
+        | Params.For<TComponentUID>;
+    };
+  };
+};
+
+type GetComponentsFromDynamicZone<
+  TSchemaUID extends Common.UID.Schema,
+  TKey extends Attribute.GetKeysByType<TSchemaUID, 'dynamiczone'>
+> = Attribute.Get<TSchemaUID, TKey> extends Attribute.DynamicZone<infer TComponentsUIDs>
+  ? TComponentsUIDs
+  : never;
+
+type ObjectNotation<TSchemaUID extends Common.UID.Schema> =
+  | {
+      [key in PopulatableKeys<TSchemaUID>]?:
+        | Boolean
+        | Params.For<
+            Attribute.GetTarget<
+              TSchemaUID,
+              key extends keyof Attribute.GetAll<TSchemaUID> ? key : never
+            >
+          >;
+    }
+  | PopulateDynamicZoneFragments<TSchemaUID>;
+
 export type Any<TSchemaUID extends Common.UID.Schema> =
   | StringNotation<TSchemaUID>
   | ArrayNotation<TSchemaUID>
-  | Object;
+  | ObjectNotation<TSchemaUID>;
```

Comment:
If the types are not generated for a project, PopulateDynamicZoneFragments resolves to {} and thus allow any value to be passed (numbers, strings, booleans, objects, ...) + the Populate.Any will resolve to any. It might be worth checking if what we've done for the filters couldn't be applied there too (see ObjectNotation)